v1/protocol_binding/binder.h

```diff
@@ -0,0 +1,157 @@
+                return set_contenttype;
+            }
+
+            absl::Status set_payload = SetPayload(msg, (*serialization).serialization);
+            if (!set_payload.ok()) {
+                return set_payload;
+            }
+        
+            return msg;
+        }
+
+        // Create CloudEvent from given Message
+        absl::StatusOr<io::cloudevents::v1::CloudEvent> Unbind(Message& message) {
+            absl::StatusOr<std::string> contenttype = GetContentType(message);
+            if (!contenttype.ok()) {
+                return contenttype.status();
+            }
+            if ((*contenttype).empty() || 
+                    (*contenttype).rfind(kContenttypePrefix.data(), 0) != 0) {
+                return UnbindBinary(message);
+            }
+
+            std::string format_str = (*contenttype).erase(
+                0, strlen(kContenttypePrefix.data()));
+
+            absl::StatusOr<cloudevents::format::Format> format;
+            format = cloudevents::formatter_util::
+                FormatterUtil::DestringifyFormat(format_str);
+            if (!format.ok()){
+                return format.status();
+            }
+
+            absl::StatusOr<std::string> get_payload = GetPayload(message);
+
+            if (!get_payload.ok()) {
+                return get_payload.status();
+            }
+
+            absl::StatusOr<std::unique_ptr<cloudevents::format::Formatter>> get_formatter;
+            get_formatter = cloudevents::formatter_util::FormatterUtil::GetFormatter(*format);
+            if (!get_formatter.ok()) {
+                return get_formatter.status();
+            }
+
+            cloudevents::format::StructuredCloudEvent structured_cloud_event;
+            structured_cloud_event.format = (*format);
+            structured_cloud_event.serialization = (*get_payload);
+
+            absl::StatusOr<io::cloudevents::v1::CloudEvent> deserialization = (*get_formatter) -> Deserialize(structured_cloud_event);
+            if (!deserialization.ok()){
+                return deserialization.status();
+            }
+            return (*deserialization);
+        }
+
+
+    // The following operations are protocol-specific and
+    // will be overriden for each supported ProtocolBinding
+    protected:
+        absl::StatusOr<std::string> GetContentType(Message& message) {
+            return absl::InternalError("Unimplemented operation");
+        }
+        
+        absl::StatusOr<std::string> GetPayload(Message& message) {
+            return absl::InternalError("Unimplemented operation");
+        }
+
+        // Marshals a BinaryContentMode message into a CloudEvent
+        absl::StatusOr<io::cloudevents::v1::CloudEvent> UnbindBinary(Message& binary_message) {
+            return absl::InternalError("Unimplemented operation");
+        }
+
+        // _____ Operations used in Bind _____
+
+        absl::Status SetContentType(Message& message, std::string contenttype) {
```

Comment:
prefer absl::string_view

Reply:
Since I will end up needing a std::string or char* anyway in order to update the message, is it alright to change it to a const ref instead?

Ex: In pubsub binder this function is overloaded to become:
```
absl::Status Binder<PubsubMessage>::SetContentType(
    PubsubMessage& pubsub_msg, std::string contenttype) {
  (*pubsub_msg.mutable_attributes())[kPubsubContentKey] = contenttype;
  return absl::Status();
}
```